packages/teuchos/comm/src/Teuchos_StackedTimer.cpp

```diff
@@ -0,0 +1,204 @@
+// @HEADER BEGIN
+// @HEADER END
+
+#include "Teuchos_StackedTimer.hpp"
+#include <limits>
+
+
+namespace Teuchos {
+
+Teuchos::RCP<StackedTimer> StackedTimer::timer;
```

Comment:
If we only need one of these global objects ever, it would be more idiomatic to use `std::unique_ptr`.  Teuchos assumes C++11 as of a few months ago.

Reply:
I was hoping this might go away. I could see a use case where we might want a separate stacked timer for a separate piece of code or separate app in a coupling. I know we could segregate by timer name, but that can be a pain.  It's more of a corner case, though and maybe not wortt the time. @bathmatt - in the original implementation I could not find anything in the stacked timer using this variable. Maybe it was meant for applications to use externally. Does anything actually use this? Could we get rid of it? If we do keep this, we could also switch TimeMonitor to use this directly.

Reply:
I'd definitely recommend removing this unless its critical to the whole system. Global variables cause lots of trouble.